server/services/educationAndWorkPlanService.ts

```diff
@@ -31,4 +31,8 @@ export default class EducationAndWorkPlanService {
     const updateGoalRequest = toUpdateGoalRequest(updateGoalDto)
     return this.educationAndWorkPlanClient.updateGoal(prisonNumber, updateGoalRequest, token)
   }
+
+  async getTimeline(prisonNumber: string, token: string): Promise<Timeline> {
```

Comment:
I think this might need some thought ....
I was envisaging that this method (`getTimeline`) would possibly be in its own service class `TimelineService`, where we have a service class designed around business domains/terminology, rather than where it gets the data from.
(the elephant in the room is that `EducationAndWorkPlanService` is clearly the wrong name, and should be split out to something like `GoalService` and `ActionPlanService` .... but that is one for another day and not this PR)

The other thing, not withstanding _where_ this method lives, is that its wrong and might fail at runtime! (or maybe not because there are no "types" in the resultant javascript 🤔 )
Anyway, its wrong because the method signature says it will return a promise of `Timeline`, where `Timeline` is your new view model.
But the `educationAndWorkPlanClient.getTimeline` method returns a promise of `TimelineResponse` , which is the API type.
If you look at all our other service methods like this, this is where we use a mapper. You'll need to write a mapper class that maps from a `TimelineResponse` and `TimelineEventResponse` into a `Timeline` and `TimelineEvent` (even if the fields are a one to one mapping, you still need a mapper)

Reply:
The thing to bear in mind here is that this mapper may not be straightforward, depending on where we apply the logic to map and filter between the API model (which is granular and includes updates to steps) and the view model (which presents a simplified view of these events, grouping some of them together).

I mentioned yesterday that it's probably worth the 3 of us chatting about this mapping/filtering, as there are some complexities/nuances here (e.g. concerning the `correlationId` that you've imported as part of the updated swagger spec).

Reply:
Wrt the `EducationAndWorkPlanService`, I agree that's not the best name and is something we should address (in a separate PR). I like the idea of naming services in terms of the business domain (so `TimelineService` would fit), rather than based on services (or "backend systems") which can change. The CIAG API is a prime example of this, though fortunately `CiagInductionService` works because CIAG is the business term for the people carrying out prisoner Inductions, and not just the name of the team who wrote the API! 😄

Reply:
You may then reasonably ask whether this means that the client class should be called `TimelineClient` or `EducationAndWorkPlanClient`? I think there's a key difference here. The client classes abstract away the details of the APIs they're calling. They know all about the API, including it's URL and data model and I think it makes sense to have one client per API (and one set of configuration for each one). 

In contrast, the service classes are more interested in the "business" domain, which in theory doesn't change if the underlying API changes (e.g. we move Inductions from the CIAG API to our API: Inductions are still the same). However, I realise this argument is flawed because our client classes are returning the API model objects to the service classes, without doing any mapping, so we can't re-write an API without it impacting the service classes. I'm not sure if we've got this right, but either way the intention is different. Hope that makes sense!